core/iwasm/aot/aot_loader.c

```diff
@@ -4235,6 +4235,12 @@ create_sections(AOTModule *module, const uint8 *buf, uint32 size,
     return false;
 }
 
+static bool
+aot_compatible_version(uint32 version)
+{
+    return version == AOT_CURRENT_VERSION || version == 3;
```

Comment:
Just checking whether I got it right 

I think this code implies that aot version 3 is compatible with everything including newer versions 

Is it something we would like to commit to?

Reply:
> aot version 3 is compatible with everything including newer versions

Not everything but only 4. So maybe this is better?
```suggestion
    return version == 4 || version == 3;
```

---

in *build_wasm_app.md*:

| WAMR version | AOT_CURRENT_VERSION | Compatible AOT version |
| ------------ | ------------------- | ---------------------- |
| next         | 4                   | 3,4                    |